tests/test_terrain.py

```diff
@@ -576,3 +576,105 @@ def test_get_terrain_attribute__out_dtype(self) -> None:
         assert self.dem.dtype != out_dtype
         assert np.dtype(slope.dtype) == out_dtype
         assert np.dtype(tpi.dtype) == out_dtype
+
+    def test_texture_shading(self) -> None:
+        """Test the texture_shading function."""
+
+        # Test with a simple DEM
+        dem_simple = np.array([[1, 1, 1], [1, 2, 1], [1, 1, 1]], dtype="float32")
+
+        # Test basic functionality
+        result = xdem.terrain.texture_shading(dem_simple, alpha=0.8)
+
+        # Check output properties
+        assert result.shape == dem_simple.shape
+        assert np.issubdtype(result.dtype, np.floating)
+        assert np.all(np.isfinite(result))  # No NaN values for simple case
+
+        # Test different alpha values
+        result_low = xdem.terrain.texture_shading(dem_simple, alpha=0.5)
+        result_mid = xdem.terrain.texture_shading(dem_simple, alpha=0.8)
+        result_high = xdem.terrain.texture_shading(dem_simple, alpha=1.5)
+
```

Comment:
Are there known values we can test here? 
For example, if the DEM is flat (e.g., `np.ones((3, 3))`), the texture shading should be a special value? (zero?)
Same for a fixed slope terrain? `np.array([[0, 0, 0], [1, 1, 1], [2, 2, 2]], dtype="float32")`

Reply:
Added the following tests to check for expected outcomes:

flat surface --> 0 everywhere
`test_texture_shading_flat_surface()` 

 planar ramp --> monotonic increase row to row, constant rows, no variation between columns
`test_texture_shading_planar_ramp()` 

adding vertical shift --> always signed and ~no effect on output
`test_texture_shading_offset_invariance_and_signed()` 

 increased alpha --> increased power in high frequencies
`test_texture_shading_spectral_shift_with_alpha()`

Reply:
Amazing, thanks!